src/CTA.WebForms2Blazor/FileConverters/CodeFileConverter.cs

```diff
@@ -42,8 +41,16 @@ public class CodeFileConverter : FileConverter
 
             // This code is set up to not break unit tests but still work for the demo
             // use code above after demo and just fix unit tests
-            _fileModel = _webFormsProjectAnaylzer.AnalyzerResult.ProjectBuildResult?.SourceFileBuildResults?
-                .Single(r => r.SourceFilePath.EndsWith(RelativePath))?.SemanticModel;
+            try
+            {
+                _fileModel = _webFormsProjectAnaylzer.AnalyzerResult.ProjectBuildResult?.SourceFileBuildResults?
+                    .Single(r => r.SourceFileFullPath.EndsWith(RelativePath))?.SemanticModel;
```

Comment:
For this line, you'll want to remove the null checks to get the exception to be thrown. `...ProjectBuildResult?.SourceFileBuildResults?
                    .Single(r => r.SourceFileFullPath.EndsWith(RelativePath))?.SemanticModel;` should become `...ProjectBuildResult.SourceFileBuildResults
                    .Single(r => r.SourceFileFullPath.EndsWith(RelativePath)).SemanticModel;` otherwise `_fileModel` will default to null, but the error message won't be recorded

Reply:
I left the null checks because we could catch an error on `.Single()` which will throw an error if it doesn't return exactly 1 result (ran into that issue during some test runs).

I don't think returning a null value the exceptional behavior here, but rather if an error is caught, we just default to null and let the user know. If you disagree, we can remove null checks so we can log those cases.

Reply:
Sounds good, we can leave this as is